application/cache/cache.go

```diff
@@ -0,0 +1,233 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/go-redis/cache/v8"
+	"github.com/go-redis/redis/v8"
+
+	"github.com/silinternational/wecarry-api/api"
+	"github.com/silinternational/wecarry-api/conversions"
+	"github.com/silinternational/wecarry-api/models"
+)
+
+var (
+	RequestsCache         *cache.Cache
+	PublicRequestsKeyName string = "public"
+)
+
+func init() {
+	ring := redis.NewRing(&redis.RingOptions{
+		Addrs: map[string]string{
+			"redis": "redis:6379",
+		},
+	})
+
+	RequestsCache = cache.New(&cache.Options{
+		Redis:      ring,
+		LocalCache: cache.NewTinyLFU(1000, time.Minute),
+		// use json in lieu of msgpack for encoding/decoding
+		Marshal:   json.Marshal,
+		Unmarshal: json.Unmarshal,
+	})
+}
+
+// Wrapper for cache write
+func CacheWrite(ctx context.Context, organization string, requestsMap interface{}) error {
+	return RequestsCache.Set(&cache.Item{
+		Ctx:   ctx,
+		Key:   organization,
+		Value: requestsMap,
+		TTL:   time.Hour,
+	})
+}
+
+// Get all requests visible to user in specified organization from cache, fetching data from database if needed
+func GetVisibleRequests(ctx context.Context, orgaziation models.Organization) ([]api.RequestAbridged, error) {
+	// get private requests
+	_, privateRequestsMap, err := getOrCreateCacheEntryPrivate(ctx, orgaziation)
+	if err != nil {
+		return nil, err
+	}
+
+	// get public requests
+	_, publicRequestsMap, err := getOrCreateCacheEntryPublic(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// compose them to get list of all requests visible to the current user
+	visibleRequestsList := make([]api.RequestAbridged, len(publicRequestsMap)+len(privateRequestsMap))
+	for _, privateRequest := range privateRequestsMap {
```

Comment:
I'm pretty sure you can just:

```go
visibleRequestsList := []api.RequestAbridged{}
visibleRequestsList = append(visibleRequestsList, privateRequestsMap...)
visibleRequestsList = append(visibleRequestsList, publicRequestsMap...)
```
If you make the list with the length of the other lists, then append the other lists to it the combined list is twice the length with the first half of the elements just zero value elements.

Reply:
Thanks for the tip, this team may make a Gopher of me yet :-)

Reply:
I was indeed having an issue with zero value elements, which initializing as a zero-length fixed. Unfortunately, I can't use the ... syntax as the list and map types don't match up.